### PR TITLE
feat: require signature from new ethereum key when rotating

### LINKED ISSUE #6271: Ethereum Key Rotation: Address new ACs
# Feature Overview

Implementing the new AC's added for ethereum-key rotation

## Specs
- [Link](https://github.com/vegaprotocol/specs/pull/1255) to spec or section within a spec

# Tasks
A checklist of the tasks that are needed to develop the feature and meet the acceptance criteria and feature test scenarios. Ideally, tasks would reflect the issues/pull requests likely to be created when developing the feature. 
- [ ]
- [ ]

# Acceptance Criteria
A list of criteria (aim for 3!) that have to be met for this feature to be accepted as 'done' by the product owner.

- It is possible to

Acceptance Criteria and Feature Test Scenarios can, in some cases, be closely related. If acceptance criteria become fully covered by feature test scenarios they can be removed leaving on acceptance criteria that can't be directly or fully proven with tests.

# Test Scenarios
Detailed scenarios (1-3!) that can be executed as feature tests to verify that the feature has been implemented as expected. We use the follow format:

GIVEN (setup/context) 
WHEN (action) 
THEN (assertion) For example...
See [here](https://github.com/vegaprotocol/vega/tree/develop/integration/) for more format information and examples.

# Additional Details (optional)
Any additional information that provides context or gives information that will help us develop the feature. 

# Examples (optional)
Code snippets from the spec for reference

# Definition of Done
>ℹ️ Not every issue will need every item checked, however, every item on this list should be properly considered and actioned to meet the [DoD](https://github.com/vegaprotocol/vega/blob/develop/DEFINITION_OF_DONE.md).

**Before Merging**
- [ ] Create relevant for [system-test](https://github.com/vegaprotocol/system-tests/issues) tickets with feature labels
- [ ] Code refactored to meet SOLID and other code design principles
- [ ] Code is compilation error, warning, and hint free
- [ ] Carry out a basic happy path end-to-end check of the new code
- [ ] All APIs are documented so auto-generated documentation is created
- [ ] All acceptance criteria confirmed to be met, or, reasons why not discussed with the engineering leadership team
- [ ] All Unit, Integration and BVT tests are passing
- [ ] Implementation is peer reviewed (coding standards, meeting acceptance criteria, code/design quality)
- [ ] Create [front end](https://github.com/vegaprotocol/token-frontend/issues) or [console](https://github.com/vegaprotocol/console/issues) tickets with feature labels (should be done when starting the work if dependencies known i.e. API changes)

**After Merging**
- [ ] Move development ticket to `Done` if there is **NO** requirement for new system-tests
- [ ] Resolve any issues with broken system-tests
- [ ] Create [documentation](https://github.com/vegaprotocol/documentation/issues) tickets with feature labels if functionality has changed, or is a new feature

### PR DESCRIPTION
**Pull request by wwestgarth**
closes #6271 

This introduces a new CLI command which can be used to send in a ethereum key rotation submission which contains a signature generated with the new etheruem key. The flow to rotate a key is now roughly:
```
# import or generate a new key in the nodewallet
vega nodewallet import

# send a rotation command with a future block height
vega rotate_eth --target-block=100 --rotate-from=old_eth_key

reload the wallet or checkpoint
vega nodewallet reload 
```

There are some other fixes too here for the ACs: 
- non-tendermint nodes rotating their ethereum key now do not generate signature bundles
- trying to rotate a second time before you've done the multisig dance with the keys from the first rotation

System test run with tests for new ACs passing:
https://jenkins.ops.vega.xyz/blue/organizations/jenkins/common%2Fsystem-tests-wrapper/detail/system-tests-wrapper/10942/tests

### CODE REVIEW

File: core/nodewallets/commander.go [hunk 1]
```diff
@@ -210,6 +210,14 @@ func wrapPayloadIntoInputData(data *commandspb.InputData, cmd txn.Command, paylo
...
 			panic("failed to wrap StateVariableProposal")
 		}
+	case txn.RotateEthereumKeySubmissionCommand:
+		if underlyingCmd, ok := payload.(*commandspb.EthereumKeyRotateSubmission); ok {
+			data.Command = &commandspb.InputData_EthereumKeyRotateSubmission{
+				EthereumKeyRotateSubmission: underlyingCmd,
+			}
+		} else {
+			panic("failed to wrap RotateEthereumKeySubmissionCommand")
+		}
```
Comment (jeremyletang) [inline core/nodewallets/commander.go:220]: Is that needed? we do not send the transaction from a running node are we?

Comment (jeremyletang) [reply core/nodewallets/commander.go:220]: oh but it's used from the command line where we instantiate a commander most likely..

File: core/validators/topology_snapshot.go [hunk 2]
```diff
@@ -144,6 +145,22 @@ func (t *Topology) serialisePendingEthereumKeyRotation() []*snapshot.PendingEthe
...
+	for _, r := range t.unresolvedEthKeyRotations {
+		ukrs = append(ukrs, &snapshot.PendingEthereumKeyRotation{
+			NodeId:     r.NodeID,
+			NewAddress: r.NewAddress,
+			Submitter:  r.SubmitterAddress,
+			OldAddress: r.OldAddress,
+		})
+	}
+
+	sort.SliceStable(ukrs, func(i, j int) bool { return ukrs[i].GetNodeId() < ukrs[j].GetNodeId() })
```
Comment (jeremyletang) [inline core/validators/topology_snapshot.go:159]: Can a node have multiple key rotation at a time again? If yes this is not gonna be enough I suppose and we'll need to sort over the nodeId an e.g the new address

**Review by jeremyletang — APPROVED** [review]

File: core/validators/topology_snapshot.go (hunk 2 above)
Comment (wwestgarth) [reply core/validators/topology_snapshot.go:159]: No they can't. This whole "unresolved rotation" check is to make sure that people can't stack up lots of ethereum key rotations and generate 100s of valid add signer bundles for themselves for 100s of different keys.

File: core/nodewallets/commander.go (hunk 1 above)
Comment (wwestgarth) [reply core/nodewallets/commander.go:220]: Haha yea I'm not sure what your question is :p but what I do know is that the commander panics here if we don't do this.

**Pull request merged**